### ISSUE TITLE
do we need to get more GHA budget for node-private?

### ISSUE THREAD

**Issue by sam-github** [nodejs/admin#510]
I know most people here can't see node-private, but its a copy of the nodejs/node repo in a private repo, used while discussing and developing fixes for embargoed issues.

Since its an exact copy of nodejs/node (except for the pre-release branches), it has the github action configuration.

And I got this over the weekend:

> You’ve used 100% of included services for GitHub Actions for the nodejs-private account.
To ensure your service isn't disrupted, make sure you increase your monthly spending limit.
Visit https://help.github.com/github/setting-up-and-managing-billing-and-payments-on-github/managing-your-spending-limit-for-github-actions to find out how.

This isn't a big deal, perhaps, maybe we don't need the actions?

If private PRs were handled via the GH advisory flow, we would be able to have private branches off the public nodejs/node repo, that became public only on release. This has been suggested a few times as a possible route, but it will take someone stepping up and doing an evaluation.

For now, this is more of a heads-up, and if anyone has suggestions, a place to discuss them.

**Comment by richardlau** [comment]
> 
> This isn't a big deal, perhaps, maybe we don't need the actions?
> 

There's some stuff that's currently only tested via actions (build/test with address sanitizers (ASAN) enabled and build/test from the source tarball).

**Comment by sam-github** [comment]
That's a good point.

So, who do we have that has some contact with GH? Maybe we can get some sponsored quota?

We must be out of quota because private repos are assumed, usually correctly, to be closed source, because I am certain that there is way more GHAs being triggered on nodejs/node than there are on the quite quiet nodejs-private/node-private repo.

**Comment by nschonni** [comment]
Not sure if you already looked at it, but the new-ish vulnerability disclosure stuff built in to GH has with private forks https://help.github.com/en/github/managing-security-vulnerabilities/creating-a-security-advisory
Not sure if it counts as private build quota, and maybe wouldn't work with the other Jenkins setup though

**Comment by sam-github** [comment]
I've seen it, but like I said

> This has been suggested a few times as a possible route, but it will take someone stepping up and doing an evaluation.

**Comment by mhdawson** [comment]
@bnb is this something you can help with?

**Comment by bnb** [comment]
Will reach out to some folks.

**Comment by bnb** [comment]
Also: I'd definitely recommend taking a look and seeing if we can optimize our usage at all just so we don't have an exponential growth problem.

**Comment by martinwoodward** [comment]
:wave: Martin from GitHub here - I'll talk to GitHub support and see if we can help.  It will probably mean upgrading the org to GitHub Enterprise plan so that you get a lot of private action minutes included. Shouldn't affect things though other than you'll see some additional features you don't have in teams.

Long term we want to make it so Security Advisories work for you so that you don't have to maintain a seperate org - would a couple of folks be up for a conversation with the Security Advisories team at GitHub to have a chat so they can see what needs to happen to make it so that you didn't need a seperate org?  If so who is the best group of folks to chat with?

**Comment by martinwoodward** [comment]
Can you check it now? - did I fix it for you?

**Comment by sam-github** [comment]
@martinwoodward thank you!

https://github.com/organizations/nodejs-private/settings/billing says we are GHE, I don't know how to check the GHA budget -- I got a notification by email, but it didn't include any links, and I just poked around for a couple minutes and didn't find anything.

For using the advisories, its mostly a matter of time, but also a matter of nervousness :-).

I was about to try once, but I had two concerns:
1. I didn't want to do it with a real security issue, because if I misunderstood how it worked or just pressend the wrong buttons, I might disclose prematurely.
.2 I ddn't want to do it with a fake security issue, because if it did end up being disclosed, it might trigger CVE assignation and all kinds of issues if it was mistaken for a real issue.

As I write this, it seems the obvious solution is to try an advisory based flow out with a fake repo, and see how it goes, take some notes, and then evaluate how well it fits our process, or what aspects of our process would have to change.

our current structure is:

- reports are submitted to hackerone
- we'd have to mirror that into an advisory on GH
- we need multiple private PRs to fix an issue, 1 per supported release line in the worst case
- I edit the issue heavily with a checklist of information about its status, links to PRs, to H1, to the nodejs.org blog post that announces it, etc, etc. There would need to be a place for this checklist in the advisory, but it doesn't need to be there when the advisory is announced, its not interesting for people who get the advisory.
- we have a meta-issue in node-private that describes the "next security release", with links to all the advisories and there state. It doesn't have to be published, but I'm not sure where it would go in the advisories workflow.

Basically, someone needs to spend a day (I expect) and do a dry-run walk-through of https://github.com/nodejs/node/blob/master/doc/guides/security-release-process.md, but using advisories, and PR a proposed update to that process that shows how the process would work with a pure-GH advisory flow.

I haven't had the time, and probably anybody who wants to volunteer could take a shot at it, but it would admittedly be much easier for someone with the private access to the previous sec release issues.
